src/Browser/LoginButton.js

```diff
@@ -1,92 +1,108 @@
-    // up to the time the button is loaded use the Span to dsiplay eventual error messages
+    // up to the time the button is loaded use the Span to display eventual 
+    // error messages
     this.loginButtonText = this.loginButtonSpan;
 
+    // bind actions dynamically to the button click
     this.loginButtonSpan.addEventListener('click', this.onClick.bind(this));
-    this.auth = auth;
-
-
-    this.onStateChange({ id: AuthStates.LOADING });
   }
 
   /**
-   * @param {Service} pryvService 
+   * Loads the style from the service info
    */
-  async loadAssets(pryvService) {
-    const assets = await pryvService.assets();
-    assets.loginButtonLoadCSS(); // can be async 
-    this.loginButtonSpan.innerHTML = await assets.loginButtonGetHTML();
+  async _loadAssets () {
+    this.loginButtonSpan.innerHTML = await this.auth.pryvService.getAssets().loginButtonGetHTML();
     this.loginButtonText = document.getElementById('pryv-access-btn-text');
-    const thisMessages = await assets.loginButtonGetMessages();
-    if (thisMessages.LOADING) {
-      this.myMessages = Messages(this.languageCode, thisMessages);
-    } else {
-      console.log("WARNING Messages cannot be loaded using defaults: ", thisMessages)
-    }
-    this.onStateChange(); // refresh messages
-    this.refreshText();
+    // State was not changed, only the button text, so reload state manually
+    this.onStateChange();
   }
 
-
-  refreshText() {
-    if (this.loginButtonText)
-     this.loginButtonText.innerHTML = this.text;
+  /**
+   * The same button can redirect, open auth popup or logout
+   */
+  onClick () {
+    if (this.auth.store.state.id === AuthStates.AUTHORIZED) {
+      this.auth.logOut();
+    } else if (this.auth.store.state.id === AuthStates.INITIALIZED) {
+      if (this.auth.settings.authRequest.returnURL) { // open on same page (no Popup) 
+        location.href = this.auth.pryvService.getAccessData().url;
+        return;
+      } else {
+        this._startLoginScreen();
+      }
+    }
   }
 
-  onClick() {
-    if (this.auth.state.action) {
-      this.auth.state.action.apply(this.auth);
+  onStateChange () {
+    this.text = this.auth.pryvService.defaultOnStateChange();
+    if (this.loginButtonText) {
+      this.loginButtonText.innerHTML = this.text;
     }
   }
 
-  onStateChange(state) {
-    if (state) {
-      this.lastState = state;
-    }
-    switch (this.lastState.id) {
-      case AuthStates.ERROR:
-        this.text = this.myMessages.ERROR + ': ' + this.lastState.message
-      break;
-      case AuthStates.LOADING:
-        this.text = this.myMessages.LOADING;
-        break;
-      case AuthStates.INITIALIZED:
-        this.text = this.myMessages.LOGIN + ': ' + this.auth.pryvServiceInfo.name;
-      break;
-      case AuthStates.AUTHORIZED:
-        this.text = this.lastState.displayName;
-        break;
-      default:
-        console.log('WARNING Unhandled state for Login: ' + this.lastState.id);
+  async _startLoginScreen () {
```

Comment:
we should adapt this as in the readme example as well https://github.com/pryv/lib-js/tree/refactor/modular-auth#using-a-custom-login-button

Reply:
do you mean, explain Login button class in more details as it was done for the custom-login-button example?

Reply:
nevermind, all good.